File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/signInCardViewResponse.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { AceData } from './aceData';
import { ActionButton } from './actionButton';
import { SignInCardParameters } from '.';
import { ICardViewResponse } from './ICardViewResponse';
import { IOnCardSelectionAction } from './IOnCardSelectionAction';

/**
 * Sharepoint SignInCardViewResponse object
 */
export class SignInCardViewResponse implements ICardViewResponse {
    private templateType: string = "SignIn";
    private aceData: AceData;
    private data: SignInCardParameters;
    private viewId: string;
    private onCardSelection: IOnCardSelectionAction;
    private cardButtons: ActionButton;

    /**
     * Initializes a new instance of the SignInCardViewResponse class
     */
    public SignInCardViewResponse() {}

    /**
     * Sets aceData property of type AceData
     */
    public set AceData(aceData: AceData) {
        this.aceData = aceData;
    }

    /**
     * Gets aceData property of type AceData
     */
    public get AceData(): AceData {
        return this.aceData;
    }

    /**
     * Sets data property of type SignInCardParameters
     */
    public set Data(data: SignInCardParameters) {
        this.data = data;
    }

    /**
     * Gets data property of type SignInCardParameters
     */
    public get Data(): SignInCardParameters {
        return this.data;
    }

    /**
     * Sets on card selection property of type IOnCardSelectionAction
     */
    public set OnCardSelection(onCardSelection: IOnCardSelectionAction) {
        this.onCardSelection = onCardSelection;
    }

    /**
     * Gets on card selection property of type IOnCardSelectionAction
     */
    public get OnCardSelection(): IOnCardSelectionAction {
        return this.onCardSelection;
    }

    /**
     * Sets card buttons property of type ActionButton
     */
    public set CardButtons(cardButtons: ActionButton) {
        this.cardButtons = cardButtons;
    }

    /**
     * Gets card buttons property of type ActionButton
     */
    public get CardButtons(): ActionButton {
        return this.cardButtons;
    }

    /**
     * Sets viewId property of type CardViewData
     */
    public set ViewId(viewId: string) {
        this.viewId = viewId;
    }

    /**
     * Gets viewId property of type CardViewData
     */
    public get ViewId(): string {
        return this.viewId;
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/SignInCardViewResponse.cs
﻿// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text;
using Microsoft.Bot.Schema.Teams;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint Primary Card View response object.
    /// </summary>
    public class SignInCardViewResponse : ICardViewResponse
    {
#pragma warning disable CA1823 // Avoid unused private fields
#pragma warning disable CS0414 // The field 'SignInCardViewResponse.templateType' is assigned but its value is never used
        private string templateType = "SignIn";
#pragma warning restore CS0414 // The field 'SignInCardViewResponse.templateType' is assigned but its value is never used
#pragma warning restore CA1823 // Avoid unused private fields

        /// <summary>
        /// Initializes a new instance of the <see cref="SignInCardViewResponse"/> class.
        /// </summary>
        public SignInCardViewResponse()
        {
        }

        /// <summary>
        /// Gets or Sets AceData for the card view of type <see cref="AceData"/>.
        /// </summary>
        /// <value>This value is the ace data of the card view response.</value>
        [JsonProperty(PropertyName = "aceData")]
        public AceData AceData { get; set; }

        /// <summary>
        /// Gets or Sets data associated with the card view of type <see cref="SignInCardParameters"/>.
        /// </summary>
        /// <value>This value is the data of the card view response.</value>
        [JsonProperty(PropertyName = "data")]
        public SignInCardParameters Data { get; set; }

        /// <summary>
        /// Gets or Sets action to be performed when card is selected of type <see cref="IOnCardSelectionAction"/>.
        /// </summary>
        /// <value>This value is the action performed when card is clicked.</value>
        [JsonProperty(PropertyName = "onCardSelection")]
        public IOnCardSelectionAction OnCardSelection { get; set; }

        /// <summary>
        /// Gets or Sets button(s) on the card view of type <see cref="ActionButton"/>.
        /// </summary>
        /// <value>This value is the button(s) associated with the card view.</value>
        [JsonProperty(PropertyName = "cardButtons")]
        public IEnumerable<ActionButton> CardButtons { get; set; }

        /// <summary>
        /// Gets or Sets the view Id of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the view id of the card view.</value>
        [JsonProperty(PropertyName = "viewId")]
        public string ViewId { get; set; }
    }
}
